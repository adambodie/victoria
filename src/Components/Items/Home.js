import React from 'react';

const Home = props => (
				<div className="home-photo">
					<img src={props.image} alt={props.alt}/>
					<div className="home-overlay">
						<h3>{props.title}</h3>
						<p>{props.text}</p>
					</div>
				</div>
			);

export default Home;
