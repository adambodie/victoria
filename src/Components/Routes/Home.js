import React from 'react';
import { Link } from "react-router-dom";

const Home = () => (
  <div>
    <div className="buttons">
		<div><Link to='/visit'>A</Link></div>
		<div><Link to='/eat'>B</Link></div>
		<div><Link to='/learn'>C</Link></div>
	</div>
	<div className="content-parent">
		<div className="content">
			<div></div>
			<div></div>
			<div></div>
			<div></div>
			<div></div>
			<div></div>
		</div>
	</div>
</div>
);

export default Home;
