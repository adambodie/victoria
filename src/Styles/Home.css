.home-overlay h3 {
	font-size: 0.8em;
	line-height: 1.0em;
	margin-top: 1.0em;
	margin-bottom: .5em;
	font-weight: bold;
}
.home-overlay p {
	font-size: .6em;
	line-height: 1.5;
	margin: 0 0 1.5em;
}
.content-parent {
	display: block;
	margin: 0 auto;
	width: 80%;
	margin-top: 30px;
}
.content {
	display: flex;
	flex-wrap: wrap;
	justify-content: space-between;
}
.buttons {
	display: flex;
	justify-content: space-around;
	margin-top: -10px;
	padding: 20px;
	box-shadow: 2px 2px 4px rgba(0,0,0, 0.5);
}

.homeButtons {
	text-align: center;
	width: 60px;
	height: 60px;
	display: flex;
	border-radius: 100%;
}

.homeButtons:hover {
	background-color: #CC1921!important;
}

.home-container {
	text-align: center;
}

.home-photo {
	position: relative;
	line-height: 0;
	margin-bottom: 2em;
	overflow: hidden;
	box-shadow: 4px 4px 8px rgba(0,0,0, 0.5);
}
.home-photo img {
 	max-width: 100%;
 	
}

/* ---- Photo Overlay ---- */

.home-overlay {
	color: #fff;
	background: rgba(0,0,0, 1);
	position: absolute;
	top: 0;
	right: 0;
	bottom: 0; 
	left: 0;
	padding-left: 20px;
	padding-right: 20px;
	display: flex; /* For centering text inside .photo-overlay */
	flex-direction: column;
	justify-content: center;
	align-items: center;
	margin-top: 50%;

}


/* ================================= 
  Media Queries
==================================== */


@media (min-width: 769px) {
	.home-container {
		width: 45%;
	}
}

/* ================================= 
  Photo Overlay Transitions
==================================== */


@media (min-width: 767px) {
	.home-overlay h3 {
		font-size: 1.4em;
		line-height: 1.25em;
		margin-top: .85em;
		margin-bottom: .5em;
	}
	.home-overlay p {
		font-size: 1em;
		line-height: 1.4;
	}
	.home-overlay {
		background: rgba(0,0,0, 0);
		opacity: 0;
		transition: opacity .5s, background .5s;
	}
	.home-overlay:hover {
		background: rgba(0,0,0, 1);
		opacity: 1;
	}

}


