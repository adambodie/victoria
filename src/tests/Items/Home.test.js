import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../../Components/Items/Home';

it('renders a feature on the home page', () => {
  const tree = renderer.create(<Home image='port-angeles' alt='Port Angeles' title='Port Angeles' text='Visit the Gateway to the Olympic National Park' />).toJSON();
  expect(tree).toMatchSnapshot();
});
