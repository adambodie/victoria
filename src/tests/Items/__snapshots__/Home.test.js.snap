// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a feature on the home page 1`] = `
<div
  className="home-photo"
>
  <img
    alt="Port Angeles"
    src="../assets/port-angeles.jpg"
  />
  <div
    className="home-overlay"
  >
    <h3>
      Port Angeles
    </h3>
    <p>
      Visit the Gateway to the Olympic National Park
    </p>
  </div>
</div>
`;
